Zend/zend_ast.c

```diff
@@ -1002,6 +1014,111 @@ ZEND_API zend_result ZEND_FASTCALL zend_ast_evaluate_inner(
 			}
 			return SUCCESS;
 		}
+		case ZEND_AST_CALL:
+		case ZEND_AST_STATIC_CALL:
+		{
+			zend_function *fptr;
+			switch (ast->kind) {
+				case ZEND_AST_CALL: {
+					ZEND_ASSERT(ast->child[1]->kind == ZEND_AST_CALLABLE_CONVERT);
+					zend_ast_fcc *fcc_ast = (zend_ast_fcc*)ast->child[1];
+					fptr = ZEND_MAP_PTR_GET(fcc_ast->fptr);
+
+					if (!fptr) {
+						zend_string *function_name = zend_ast_get_str(ast->child[0]);
+						zend_string *function_name_lc = zend_string_tolower(function_name);
+						fptr = zend_fetch_function(function_name_lc);
+						if (!fptr && ast->child[0]->attr != ZEND_NAME_FQ) {
+							const char *backslash = zend_memrchr(ZSTR_VAL(function_name_lc), '\\', ZSTR_LEN(function_name_lc));
+							if (backslash) {
+								fptr = zend_fetch_function_str(backslash + 1, ZSTR_LEN(function_name_lc) - (backslash - ZSTR_VAL(function_name_lc) + 1));
```

Comment:
```suggestion
								fptr = zend_fetch_function_str(backslash + 1, ZSTR_LEN(function_name_lc) - (backslash - ZSTR_VAL(function_name_lc)) - 1);
```

I find this slightly more logical, but ignore if you disagree.

Reply:
I agree, but I had taken this from:

https://github.com/php/php-src/blob/35be6d071567176c7bd411afeaec152af618b208/ext/reflection/php_reflection.c#L3538

So I'll keep this for consistency. It would be great to have namespaces as first-class citizens …